### DISCUSSION TITLE
Why does linaria not work in the app directory, but normal CSS modules do?

### DISCUSSION THREAD

**Discussion by dlehmhus** [vercel/next.js#42253] [Help]
First of all, I have read the [part](https://beta.nextjs.org/docs/styling/css-in-js) in the Next.js docs, where it says linaria has not yet added support for the app directory. But I'm wondering what is missing, and why it is working for the `pages` but not for the `app` directory. As far as I understood it, linaria turns my css-in-js with the help of webpack at build time into CSS modules. And since normal CSS modules [work](https://beta.nextjs.org/docs/styling/css-modules) insight the app dir, I'm wondering now what the issue might be.

### A quick example:
The [linaria webpack loader](https://github.com/callstack/linaria/tree/master/packages/webpack5-loader) turns this:
```tsx
// app/page.tsx
import { styled } from '@linaria/react'

const Container = styled.div`
  background: fuchsia;
` 

export default function Home() {
  return (
    <Container>
      Hello App Dir
    </Container>
  )
}
```
into this: 
```tsx
const Container = /*#__PURE__*/styled('div')({
  name: "Container",
  class: "c1ypg2vm",
  propsAsIs: false
});
export default function Home() {
  return <Container>
      Hello App Dir
    </Container>;
}

require("./page.linaria.module.css!=!../../linaria/packages/webpack5-loader/lib/outputCssLoader.js?cacheProvider=!./page.tsx");
```
In my mind, that is now a CSS module (okay, with an inline loader, but that works fine for pages...). What am I missing? Any pointers would be greatly appreciated!

**Comment by wwwdepot** [comment]
See https://beta.nextjs.org/docs/styling/css-in-js

**Reply by dlehmhus** [reply]
Hi @wwwdepot, what do I see there? Is there a specific part that I missed? As I mentioned in my first sentence, I am aware that linaria is not working. But my question here is, why? What is missing?
